### DISCUSSION TITLE
Diffused spheres don't look as bright as they are in the book.

### DISCUSSION THREAD

**Discussion by seanballais** [RayTracing/raytracing.github.io#874] [Debugging]
I'm loving the ray tracing book, but I'm currently stuck at the part on diffused reflections. The book shows a rendered output with no gamma correction that is light grey. However, my output is far from it as shown below.

![My output (no gamma correction added).](https://user-images.githubusercontent.com/7175885/111875833-be153680-89d6-11eb-9067-323e0985b25e.png)

Compare my output to that of the output from the book:

![The book's output (no gamma correction added).](https://user-images.githubusercontent.com/7175885/111875873-f74da680-89d6-11eb-8b76-5c4689b18f37.png)

With gamma correction added, my output became lighter.

![My output (with gamma correction added).](https://user-images.githubusercontent.com/7175885/111875925-2e23bc80-89d7-11eb-83e3-08dc600819a4.png)

But, the book's output is even lighter.

![The book's output (with gamma correction added).](https://user-images.githubusercontent.com/7175885/111875951-4bf12180-89d7-11eb-8a6a-e3ae0c006d44.png)

I followed the code, but modified the structure to something I find better, so I'm not sure where I made a mistake. My current guess is that it has something to do with the way the sky is rendered.

Any tips that can help me solve my problem? Thanks!

**Comment by ChuWesly** [comment]
I have the same problem. I also made some very small adjustments to the code with only in regards in outputting it in a ppm file in the program itself. This shouldn't change anything.

**Reply by ChuWesly** [reply]
I think it has to do with some kind of floating point implementation if the t is smaller than 0. If you go to the next paragraph with shadow acne the image rapidly changes. Below the image with shadow correction and gamma correction. It is also way faster with the acne fix.


![image](https://user-images.githubusercontent.com/64785006/111907052-f149db00-8a53-11eb-9842-dcc13cc9ceb1.png)

**Comment by tysonliddell** [comment]
See https://github.com/RayTracing/raytracing.github.io/issues/1013 and https://github.com/RayTracing/raytracing.github.io/issues/875. Underlying cause appears to be [this bug](https://github.com/RayTracing/raytracing.github.io/issues/875#issuecomment-1013842362).
